cms/djangoapps/contentstore/views/item.py

```diff
@@ -1052,7 +1058,7 @@ def _get_gating_info(course, xblock):
         if not hasattr(course, 'gating_prerequisites'):
             # Cache gating prerequisites on course module so that we are not
             # hitting the database for every xblock in the course
-            setattr(course, 'gating_prerequisites', gating_api.get_prerequisites(course.id))
+            course.gating_prerequisites = gating_api.get_prerequisites(course.id)
```

Comment:
Do you know why we ever used `setattr` instead of doing it the way you have here? Your way is definitely clearer, but I wonder if there was any weird magic going on there.

Though I guess as long as the tests pass, I shouldn't be so paranoid.

Reply:
I don't know why `setattr` was used. I changed it because pylint complained.